### DISCUSSION TITLE
Migration path from Enterprise instance of Stack Overflow

### DISCUSSION THREAD

**Discussion by garyniemen** [drodil/backstage-plugin-qeta#147] [General]
Curious if there is a migration path from an Enterprise instance of Stack Overflow. Or, if anybody has done something similar.

**Comment by drodil** [comment]
There's some discussion about this at https://github.com/drodil/backstage-plugin-qeta/issues/126 

But basically, there's no supported way to migrate from another system as of now. It would be nice to get some scripts or CLI tool for the migration now that it's possible to call the Backstage APIs with static API tokens: https://github.com/backstage/backstage/pull/24257

I cannot participate in this work much as I don't have access to any systems like StackOverflow to do migration from so it's pretty hard to test things. Of course I can try to help as much as possible if someone takes this challenge :)

**Reply by garyniemen** [reply]
If we go down this route, it is likely we will have a go at this. I'll let you know.

**Reply by drodil** [reply]
@garyniemen just wanted to ping you and ask what was your course of action? You might have noticed that this plugin has gathered quite a lot of new features like articles, following tags/entities/users, AI support etc. 

Unfortunately, I am still unable to support the migration from SO but just reaching out to see what is going on? :)

**Comment by garyniemen** [comment]
Hi @drodil - after discussion, we decided to prioritise implementing a RAG-based feature on top of TechDocs (instead of adding the q&a Backstage plugin and importing the SO data).

**Reply by drodil** [reply]
Cool idea :) Hope it works for you!

**Reply by FullStackRanger** [reply]
> Hi @drodil - after discussion, we decided to prioritise implementing a RAG-based feature on top of TechDocs (instead of adding the q&a Backstage plugin and importing the SO data).

Were you succesful with this, we're doing something similar and I was looking for a common pattern to compare.

**Comment by kirbarn** [comment]
So, we went with this approach and successfully migrated the data.

StackExchange offers a script that allows you to export all posts, articles and some other data: https://github.com/StackExchange/so4t_data_export

We then created another script that would inject the data into the Q&A DB with SQL statements with some custom filtering. We linked questions to existing entities from the catalog based on a mapping that we created manually - mapped tags from Stack to entities from Backstage (wasn't a lot of work). Where we couldn't match the original author with a Backstage user, we would set the author to user:default/anonymous. Note that you want to run this script on an empty DB unless you want to deal with the hassle of reconciling relationships with existing tags and entities.

We migrated around 5k questions this way. Everything worked perfectly.

**Reply by ScottGuymer** [reply]
Any chance you could share the script you used to inject the data?
